src/Test/Utilities/Portable/TestBase.cs

```diff
@@ -237,7 +237,10 @@ public static MetadataReference MscorlibRef
             {
                 if (s_mscorlibRef == null)
                 {
-                    s_mscorlibRef = AssemblyMetadata.CreateFromImage(TestResources.NetFX.v4_0_30319.mscorlib).GetReference(display: "mscorlib.v4_0_30319.dll");
+                    Interlocked.CompareExchange(
+                        ref s_mscorlibRef,
+                        AssemblyMetadata.CreateFromImage(TestResources.NetFX.v4_0_30319.mscorlib).GetReference(display: "mscorlib.v4_0_30319.dll"),
+                        null);
```

Comment:
@cston Sorry for the delay, I forgot this PR was still open -- but it's ready for re-review when you get a chance.